python/ray/data/datasource/parquet_datasource.py

```diff
@@ -190,12 +190,14 @@ def __init__(
         self._schema = schema
 
     def estimate_inmemory_data_size(self) -> Optional[int]:
-        # TODO(ekl) better estimate the in-memory size here.
-        PARQUET_DECOMPRESSION_MULTIPLIER = 5
+        # TODO(ekl/chengsu) better estimate the in-memory size here,
+        # when columns pruning is used.
         total_size = 0
-        for meta in self._metadata:
-            total_size += meta.serialized_size
-        return total_size * PARQUET_DECOMPRESSION_MULTIPLIER
+        for file_metadata in self._metadata:
+            for row_group_idx in range(file_metadata.num_row_groups):
+                row_group_metadata = file_metadata.row_group(row_group_idx)
+                total_size += row_group_metadata.total_byte_size
+        return total_size
 
```

Comment:
I think we should also change the size reporting for the metadata in other locations. Currently, the size_bytes() call returns the same incorrect information.

Reply:
It may be still not accurate then. A user running on 1.11 release mentioned a few times the in-memory size estimation of Parquet is off with size_bytes(). But we can double check some examples.

Reply:
> It may be still not accurate then. A user running on 1.11 release mentioned a few times the in-memory size estimation of Parquet is off with size_bytes(). But we can double check some examples.

By off, we mean the estimation is much smaller, or larger than actual size? Curious to learn more, but the estimation before this PR is far more inaccurate.

Reply:
I tried locally and the estimation is slightly larger than the real size, so it seems reasonable. Could it be the user with 1.11 was using an incorrect version from before?

Reply:
Yeah `size_bytes()` shouldn't have this same bug, at least.

>It may be still not accurate then. A user running on 1.11 release mentioned a few times the in-memory size estimation of Parquet is off with size_bytes(). But we can double check some examples.
>By off, we mean the estimation is much smaller, or larger than actual size? Curious to learn more, but the estimation before this PR is far more inaccurate.

It's a known UX issue that the post-read `ray.data.read().size_bytes()` calls are an underestimation since it's returning the file size, not the in-memory size. Using the Parquet inflation multiplier for best-effort parallelism determination makes sense, but I don't know if we should be exposing it to the user since it (1) won't always be very accurate and (2) that potential inaccuracy is harder to document than the current state, which is a bit easier to explain: on-disk size post-read, call `.fully_executed()` to read all of the data and get the real size.

We should open a separate issue for this and discuss what estimate makes the most sense!